### PR TITLE
Forum username validation and normalisation

### PR DESCRIPTION
**Pull request by arnaud-morvan**
Relate https://github.com/c2corg/v6_forum/issues/22

TODO : Update the V5 forum username normalisation script.

### CODE REVIEW

**Comment by arnaud-morvan** [comment]
@tsauerwein : you can take a look at this as I'm working on normalisation script.

File: c2corg_api/models/user.py [hunk 1]
```diff
@@ -153,6 +153,53 @@ def validate_password(self, plain_password):
       unique=True)
 
 
+# Check forum_username validity with discourse
+# https://github.com/discourse/discourse/blob/master/app/models/username_validator.rb
+check_forum_username_ddl = DDL("""
+CREATE OR REPLACE FUNCTION users.check_forum_username(name TEXT)
```
Comment (tsauerwein) [inline c2corg_api/models/user.py]: Have you tested the trigger? Or did you consider adding a unit-test for the trigger?

Comment (arnaud-morvan) [inline c2corg_api/models/user.py]: I've made some basic tests, not sure to have time to add tests for the check constraint.

**Comment by arnaud-morvan** [comment]
I've opened an ticket for the forum_username check constraint tests with label "extra".

**Comment by tsauerwein** [comment]
@arnaud-morvan I tried to run a migration (with the v5 database on the dev. server) and it failed with:

```
sqlalchemy.exc.IntegrityError: (psycopg2.IntegrityError) duplicate key value violates unique constraint "ix_users_user_lower_forum_username"
DETAIL:  Key (lower(forum_username::text))=(laurentd1) already exists.
```

Are there changes required to the script `create_unique_normalized_usernames.sql`?

**Comment by arnaud-morvan** [comment]
These changes are made in this pull request.

**Pull request merged**